### ISSUE TITLE
How to access the updated structure from Update()?

### ISSUE THREAD

**Issue by jhvst** [dancannon/gorethink#126]
Hello,

I was wondering how do you access the updated structure from Update() response? I've tried to use Update() with UpdateOpts parameter and without, but can't find a way to get satisfying response. Cursor only gives me one result, which doesn't populate the other fields anyway. Using RunWrite() instead of Run() gives me WriteResponse, but I can't find a way to unmarshal the NewValue from Changes either.

I made my own tests samples at https://github.com/9uuso/gorethink/tree/update-bug, which check the response in addition to just making the queries. Both of the Update queries however fail, so maybe you could take a look at it and point out what I'm doing wrong?

**Comment by dancannon** [comment]
Currently prioritising #125 and v1.14 release. Will look at this soon I promise!

**Comment by dancannon** [comment]
Hi,

So as you noticed RunWrite provides a general map based structure, if you want to unmarshal the value of the changes field you will need to create your own WriteResponse type. Also you cannot just directly unmarshal the response of Insert/Update into your type. You will need to use the Changes fields.

Also in your tests you had incorrect struct tags (both fields had the same name) which looked to be breaking things which seems to be another issue that I will fix shortly.

Here is an example based on your tests https://gist.github.com/dancannon/88f2283ad6a6d181c2ab

Hope this helps,
Daniel

**Comment by jhvst** [comment]
Hi again,

Thank you for your comprehensive response and the example in specific! I was able to fix my issue with the help of the gist you provided. I knew I was doing something wrong when I couldn't find a way to use the `Changes[0].NewValue` the way I wanted. I'm glad the fix was not all too complicated either.

And indeed, I unintentional named the structure tags the same. 

The issue is now closed for my part. Cheers.
